[Linux] build fails due to empty -l flags

I tried building Xenia on Fedora 34. With workaround for #1825 I was making good progress. Unfortunately, I hit the following issue a few times:
```
/usr/bin/ld: cannot find -l-lstdc++fs
```
This comes from
```
build/xenia-ui.make:LIBS += bin/Linux/Release/libxenia-base.a -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -l -lstdc++fs -ldl -llz4 -lpthread -lrt
```
There is an `-l` between `-lglib-2.0` and `-lstdc++fs`. In premake.lua there appears to be nothing wrong:
```
filter("platforms:Linux")
  system("linux")
  toolset("clang")
  buildoptions({
    -- "-mlzcnt",  -- (don't) Assume lzcnt is supported.
  })
  pkg_config.all("gtk+-x11-3.0")
  links({
    "stdc++fs",
    "dl",
    "lz4",
    "pthread",
    "rt",
  })
```
Any ideas where the `-l` comes from?

--- a/tools/build/scripts/pkg_config.lua
+++ b/tools/build/scripts/pkg_config.lua
@@ -24,7 +24,9 @@ function pkg_config.lflags(lib)
   local output = ({os.outputof("pkg-config --libs-only-l "..lib)})[1]
   for k, flag in next, string.explode(output, " ") do
     -- remove "-l"
-    links(string.sub(flag, 3))
+    if flag ~= "" then
+        links(string.sub(flag, 3))
+    end
   end
 end
 

